src/commands/mod.rs

```diff
@@ -391,6 +400,91 @@ impl DaemonControl {
+            input: Vec::with_capacity(coins_outpoints.len()), // Will be zero capacity for coin selection.
+            output: Vec::with_capacity(destinations.len()),
+        };
+        // Add the destinations outputs to the transaction and PSBT. At the same time record the
+        // total output value to later compute fees, and sanity check each output's value.
+        let mut out_value = bitcoin::Amount::from_sat(0);
+        let mut psbt_outs = Vec::with_capacity(destinations.len());
+        for (address, value_sat) in destinations {
+            let address = self.validate_address(address.clone())?;
+
+            let amount = bitcoin::Amount::from_sat(*value_sat);
+            check_output_value(amount)?;
+            out_value = out_value.checked_add(amount).unwrap();
+
+            tx.output.push(bitcoin::TxOut {
+                value: amount.to_sat(),
+                script_pubkey: address.script_pubkey(),
+            });
+            // If it's an address of ours, signal it as change to signing devices by adding the
+            // BIP32 derivation path to the PSBT output.
+            let bip32_derivation =
+                if let Some((index, is_change)) = db_conn.derivation_index_by_address(&address) {
+                    let desc = if is_change {
+                        self.config.main_descriptor.change_descriptor()
+                    } else {
+                        self.config.main_descriptor.receive_descriptor()
+                    };
+                    // NOTE: include all derivations for change outputs, to make sure signers are
+                    // able to re-compute change addresses.
+                    desc.derive(index, &self.secp).bip32_derivations(|_| true)
+                } else {
+                    Default::default()
+                };
+            psbt_outs.push(PsbtOut {
+                bip32_derivation,
+                ..PsbtOut::default()
+            });
+        }
+        assert_eq!(tx.output.is_empty(), is_self_send);
+        // Get the change address to create a dummy change txo.
+        let change_index = db_conn.change_index();
+        let change_desc = self
+            .config
+            .main_descriptor
+            .change_descriptor()
+            .derive(change_index, &self.secp);
+        let mut change_txo = bitcoin::TxOut {
+            value: std::u64::MAX,
+            script_pubkey: change_desc.script_pubkey(),
+        };
+        // If change output is indeed required, the next change index in DB will be updated below.
+        let mut selected_change = bitcoin::Amount::from_sat(0); // For coin selection, will tell us how much change required.
+        let selected_coins_outpoints: Vec<bitcoin::OutPoint>;
+        let final_coins_outpoints: &[bitcoin::OutPoint] = if use_coin_selection {
+            log::info!("No outpoints specified. Selecting coins...");
+            let candidate_coins: Vec<Coin> = db_conn
+                .coins(&[CoinStatus::Confirmed], &[])
```

Comment:
I think I should also include `CoinStatus::Unconfirmed` here to be consistent with the coins available for selection in the GUI.